java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture4/tasks/_4/Number4.java

```diff
@@ -22,110 +22,58 @@ enum Weight {
-            return ((mass * 453.6) + " " + GRAM);
-        } else if (sourceUnit == POUND && convertTo == KILOGRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 2.205) + " " + KILOGRAM);
-        } else if (sourceUnit == POUND && convertTo == OUNCE) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 16) + " " + OUNCE);
-        } else if (sourceUnit == POUND && convertTo == TON) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 2205) + " " + TON);
-
-            //MARK: TON
-        } else if (sourceUnit == TON && convertTo == GRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 1000000) + " " + GRAM);
-        } else if (sourceUnit == TON && convertTo == KILOGRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 1000) + " " + KILOGRAM);
-        } else if (sourceUnit == TON && convertTo == OUNCE) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 35270) + " " + OUNCE);
-        } else if (sourceUnit == TON && convertTo == POUND) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 2205) + " " + POUND);
-        }
-        return "ERROR";
-    }
-
-    static Weight enumValueByAString(String string) {
-        switch (string) {
-            case "GRAM", "Gram", "gram", "gr" -> {
-                return GRAM;
+    public float convertValue(float mass, Weight sourceUnit) {
+        switch (sourceUnit) {
+            case GRAM -> {
+                System.out.println("Converting Gram to Kilogram...");
+                return mass / 1000;
             }
-            case "KILOGRAM", "Kilogram", "kilogram", "kg" -> {
-                return KILOGRAM;
+            case KILOGRAM -> {
+                System.out.println("Converting Kilogram to Ounce...");
+                return mass * 35.274F;
             }
-            case "OUNCE", "Ounce", "ounce", "oz" -> {
-                return OUNCE;
+            case OUNCE -> {
+                System.out.println("Converting Ounce to Pound...");
+                return mass * 2.205F;
             }
-            case "POUND", "Pound", "pound", "lb" -> {
-                return POUND;
+            case POUND -> {
+                System.out.println("Converting Pounds to Ton...");
+                return mass / 2205F;
             }
-            case "TON", "Ton", "ton", "t" -> {
-                return TON;
+            case TON -> {
+                System.out.println("Converting Ton to Pounds...");
+                return mass * 2205F;
             }
         }
-        return GRAM;
+        return mass;
+    }
+
+    public Weight enumValueByAString(String string) {
+        if (string.equalsIgnoreCase("GRAM")) {
```

Comment:
you can use the switch statement if you convert incoming string to some specific case (using `toLowerCase` or `toUpperCase`)